services/graphql/src/routes/reporting.js

```diff
@@ -0,0 +1,93 @@
+const { Parser } = require('json2csv');
+const { ObjectId } = require('mongodb');
+const Reporting = require('../services/reporting');
+const {
+  publisher, deployment, adunit, advertiser,
+  order, lineitem, ad,
+} = require('../mongoose/models');
+
+const { log } = console;
+const { isArray } = Array;
+
+const asyncRoute = fn => (req, res, next) => Promise.resolve(fn(req, res, next)).catch(next);
+const retrieve = (model, $in) => model.find({ _id: { $in } }, ['name', 'size']);
+const getModelValue = (field, row, models) => {
+  try {
+    const [model, subfield] = field.split('.');
+    const found = models[`${model}s`].filter(({ _id }) => `${_id}` === `${row[`${model}Id`]}`);
+    return found[0][subfield];
```

Comment:
This should check that `found` is defined. e.g. `if (!found) return null` or something similar.

Reply:
It's explicitly returning null after the catch.

Reply:
Not sure I like using the try/catch this way. Now errors will be logged when the filter fails... which isn't what the catch should do here. It should only log network/query errors, not logic errors that could've been avoided. :)

Reply:
In fact, shouldn't the entire request fail if there's a network error?

Reply:
Yes, the request will fail if there's a network error. I was attempting to catch if there was an orphaned reference somehow, but there's no need. The request will fail anyway if it's a null or undefined value since `row.advertiser` is not nullable in the definition.

I've removed the try/catch and error logging.